From b79e9ebf1c9a905ec88df976dc8f7ca2e98827dc Mon Sep 17 00:00:00 2001
From: zhaochengyi <zhaochengyi@uniontech.com>
Date: Thu, 3 Nov 2022 10:39:07 +0800
Subject: [PATCH] profile-set: Add Sugon conf to adapt usb audio card
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add 22d4-us-usb-audio.conf to adapt the usb audio card
which it is used in Sugon W330-H30A

Log: Sugon W330-H30A设备的声卡不是通用的usb声卡，使用pulseaudio
的通用模板无法进行正常识别，需要udev识别该usb声卡后，加载
22d4-us-usb-audio.conf进行适当配置。
Bug: https://pms.uniontech.com/zentao/bug-view-38155.html

Author: Chengyi Zhao <zhaochengyi@uniontech.com>
Origin: https://gerrit.uniontech.com/plugins/gitiles/base/pulseaudio/+/f75de2f53bc4874a99677fcd4e0bdc4165c9f666
---
 src/modules/alsa/90-pulseaudio.rules          |  2 +
 .../mixer/profile-sets/22d4-us-usb-audio.conf | 58 +++++++++++++++++++
 2 files changed, 60 insertions(+)
 create mode 100644 src/modules/alsa/mixer/profile-sets/22d4-us-usb-audio.conf

diff --git a/src/modules/alsa/90-pulseaudio.rules b/src/modules/alsa/90-pulseaudio.rules
index a74f3e6..bca8322 100644
--- a/src/modules/alsa/90-pulseaudio.rules
+++ b/src/modules/alsa/90-pulseaudio.rules
@@ -166,6 +166,8 @@ ATTRS{idVendor}=="1395", ATTRS{idProduct}=="00a1", ENV{PULSE_PROFILE_SET}="sennh
 
 # HUAWEI T100
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4987", ENV{PULSE_PROFILE_SET}="st2557-pc-usb-audio.conf"
+# Sugon W330-H35A
+ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="491b", ENV{PULSE_PROFILE_SET}="22d4-us-usb-audio.conf"
 
 GOTO="pulseaudio_end"
 
diff --git a/src/modules/alsa/mixer/profile-sets/22d4-us-usb-audio.conf b/src/modules/alsa/mixer/profile-sets/22d4-us-usb-audio.conf
new file mode 100644
index 0000000..3ae16ba
--- /dev/null
+++ b/src/modules/alsa/mixer/profile-sets/22d4-us-usb-audio.conf
@@ -0,0 +1,58 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+[General]
+auto-profiles = yes
+
+[Mapping analog-stereo-front-mic]
+description = Front Micphone
+paths-input = analog-input analog-input-front-mic
+device-strings = hw:%f,0
+channel-map = left,right
+direction = input
+
+[Mapping analog-stereo-headphone]
+description = Front Headphone
+paths-output = analog-output analog-output-headphones
+device-strings = hw:%f,0
+channel-map = left,right
+direction = output
+
+[Mapping analog-stereo-linein]
+description = Rear Linein
+paths-input = analog-input analog-input-linein
+device-strings = hw:%f,2
+channel-map = left,right
+direction = input
+
+[Mapping analog-stereo-rear-mic]
+description = Rear Microphone
+paths-input = analog-input analog-input-rear-mic
+device-strings = hw:%f,1
+channel-map = left,right
+direction = input
+
+[Mapping analog-stereo-speaker]
+description = Rear Speaker
+paths-output = analog-output analog-output-speaker
+device-strings = hw:%f,1
+channel-map = left,right
+direction = output
+
+[Profile output:analog-stereo-headphone+output:analog-stereo-speaker+input:analog-stereo-all]
+description = 22D4-US
+output-mappings = analog-stereo-headphone analog-stereo-speaker
+input-mappings = analog-stereo-front-mic analog-stereo-linein analog-stereo-rear-mic
+skip-probe = yes
-- 
2.33.1

